### PR TITLE
Prevent CI from masking version problem bugs

### PR DESCRIPTION
**Pull request by oflatt**
The current CI hides version problems by overwriting the version of Cedar used.
This PR removes that functionality.

### CODE REVIEW

File: .github/workflows/build_rust_hello_world_reusable.yml [hunk 1]
```diff
@@ -24,10 +24,6 @@ jobs:
         with:
           repository: cedar-policy/cedar-examples
           ref: ${{ inputs.cedar_examples_ref }}
-# If we passed a branch for cedar-policy, modify Cargo.toml to use that branch
-      - name: Replace Crates dot IO with Github version
-        if: "${{ inputs.cedar_policy_ref != '' }}"
-        run: cd cedar-rust-hello-world && printf "\n[patch.crates-io]\ncedar-policy = { git = 'https://github.com/cedar-policy/cedar.git', rev = '${{ inputs.cedar_policy_ref }}'}" >> Cargo.toml
```
Comment (john-h-kastner-aws) [inline .github/workflows/build_rust_hello_world_reusable.yml:30]: I agree that we should fix CI to detect a mistake in versioning, but I think this code is necessary. 
The intent of this step is that we should be able to submit a PR to `cedar-policy` and then check that `cedar-examples` builds and passes tests with the changes in PR branch, so just building against the crates.io release won't work.

Maybe try removing

https://github.com/cedar-policy/cedar-examples/blob/f94b5df835922f55a6e8327f69ced555afed9a82/.github/workflows/ci.yml#L59

https://github.com/cedar-policy/cedar-examples/blob/f94b5df835922f55a6e8327f69ced555afed9a82/.github/workflows/ci.yml#L67

and

https://github.com/cedar-policy/cedar-examples/blob/f94b5df835922f55a6e8327f69ced555afed9a82/.github/workflows/ci.yml#L75

instead.

Comment (shaobo-he-aws) [reply .github/workflows/build_rust_hello_world_reusable.yml:30]: I thought the convention is to use git as the source in cargo.toml files. So this replacement is unnecessary.

Comment (john-h-kastner-aws) [reply .github/workflows/build_rust_hello_world_reusable.yml:30]: But it needs to point to the correct git branch for a PR, not just the git `main` or release branch

**Review by andrewmwells-amazon — CHANGES_REQUESTED** [review]
We don't want to do this.

File: .github/workflows/build_rust_hello_world_reusable.yml (hunk 1 above)
Comment (andrewmwells-amazon) [reply .github/workflows/build_rust_hello_world_reusable.yml]: We don't want to do this.
The *reusable* tasks take in two branch names, one for `cedar-policy` and one for `cedar-examples`. We want this so a change to `cedar-policy:releases/3.2.x` can build against the `3.2` branch in `cedar-examples`.
For changes to `cedar-examples`, the `ci.yaml` file can point to the correct `cedar_policy_ref`.

**Comment by andrewmwells-amazon** [comment]
Let's use https://github.com/cedar-policy/cedar-examples/pull/149

**Pull request closed without merging**